lib/actionizer.rb

```diff
@@ -17,7 +17,9 @@ def method_missing(method_name, *args, &block)
 
       if instance.respond_to?(method_name)
         error = defined_inputs.check_for_param_error(method_name, *args)
-        raise ArgumentError, error if error
+        if error
+          return Actionizer::Result.new(error: error).tap(&:fail)
```

Comment:
I like this approach much better. Helps prevent exceptions from being used for control flow.